lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2170,6 +2173,13 @@ class ShellCommand : public ExternalCommand {
         return false;
       }
       inheritEnv = value == "true";
+    } else if (name == "working-directory") {
+#if __APPLE__
+      workingDirectory = value;
+#else
+      ctx.error("working-directory unsupported on this platform");
```

Comment:
Ah, I see you put this check here. In that case, I'd just add an assert for non-Apple platforms.